landscape/client/manager/shutdownmanager.py

```diff
@@ -58,9 +58,31 @@ def _respond_success(self, data, operation_id):
             lambda _: self.registry.broker.stop_exchanger())
         return deferred
 
-    def _respond_failure(self, failure, operation_id):
+    def _respond_failure(self, failure, operation_id, reboot):
         logging.info("Shutdown request failed.")
-        return self._respond(FAILED, failure.value.data, operation_id)
+        failure_report = '\n'.join([
+            failure.value.data,
+            "",
+            "Attempting to force %(operation)s. Please note that if this "
+            "succeeds, Landscape will have no way of knowing and will still "
+            "mark this activity as having failed. It is recommended you check "
+            "the state of the machine manually to determine whether "
+            "%(operation)s succeeded." %
+            {"operation": "reboot" if reboot else "shutdown"}
+        ])
+        deferred = self._respond(FAILED, failure_report, operation_id)
+        # Add another callback spawning the poweroff or reboot command (which
+        # seem more reliable in aberrant situations like a post-trusty release
+        # upgrade where upstart has been replaced with systemd). If this
+        # succeeds, we won't have any opportunity to report it and if it fails
+        # we'll already have responded indicating we're attempting to force
+        # the operation so either way there's no sense capturing output
+        protocol = ProcessProtocol()
+        command, args = self._get_command_and_args(protocol, reboot, True)
+        deferred.addCallback(
+            lambda _: self._process_factory.spawnProcess(
+                protocol, command, args=args))
+        return deferred
```

Comment:
I like that you still report it as a failure (it was). It's weird if it did succed, but there was an anomaly worth checking and the message is pretty clear.